simple/stats/cache.py

```diff
@@ -36,6 +39,25 @@ def generate_svg_cache(db: Db,
   pass
 
 
+def gzip_and_base64_encode(data: bytes) -> str:
+  """Compresses bytes using GZIP, base64 encodes them and returns the encoded string."""
+  compressed_buffer = io.BytesIO()
+
+  with gzip.GzipFile(fileobj=compressed_buffer, mode="wb") as gz_file:
+    gz_file.write(data)
+
+  return base64.b64encode(compressed_buffer.getvalue()).decode('utf-8')
+
+
+def base64_decode_and_gunzip(encoded_data: str) -> bytes:
+  """Decodes a Base64 string, decompresses the GZIP data and returns the uncompressed bytes."""
+
+  compressed_data = base64.b64decode(encoded_data)
+
+  with gzip.GzipFile(fileobj=io.BytesIO(compressed_data), mode="rb") as gz_file:
+    return gz_file.read()
```

Comment:
nit: maybe worth having this in a utils file.  TODO is fine.

Reply:
Added TODO.